python/revng/scripting/projects/daemon_project.py

```diff
@@ -0,0 +1,174 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import json
+from typing import Dict, List
+
+import yaml
+from gql import Client, gql
+from gql.transport.aiohttp import AIOHTTPTransport
+
+from .project import Project
+
+from revng.model import Binary, Function, dump_diff_to_string, load_model_from_string, make_diff  # type: ignore[attr-defined] # noqa: E501 # isort: skip
+
+
+class DaemonProject(Project):
+    """
+    This class is used to run revng analysis and artifact through the
+    revng daemon. This class does not run the daemon, this is only
+    the client that uses `gql` to forward the requests to the daemon.
+    """
```

Comment:
Too specific, drop the `gql` part, just mention that it talks to the daemon via GraphQL